ui/src/runtime/composables/useEvents.js

```diff
@@ -0,0 +1,28 @@
+export default function() {
+  const listeners = {};
+
+  function on(name, id, callback) {
+    listeners[name] ||= {};
+
+    listeners[name][id] = callback;
+  }
+
+  function off(name, id) {
+
+    delete listeners[name][id];
+  }
+
+  async function emit(name) {
+    if(!listeners[name]) { return; }
+
+    for(const id in listeners[name]) {
+      listeners[name][id]();
+    }
+  }
+
+  return {
+    on,
+    off,
+    emit
+  };
```

Comment:
let's wrap this into a module and return `create` function from this composable, so that anyone that uses `useEvents` understands the fact that they're creating a new `Events` object, not reusing an existing (page scoped) instance.